application/__tests__/example.macro.test.js

```diff
@@ -0,0 +1,174 @@
+    expect(codeExampleTabsContainer.nodeName.toLowerCase()).toBe('ul')
+
+    const exampleToggleLinks = codeExampleTabsContainer.querySelectorAll('li.app-tabs__item a')
+    expect(exampleToggleLinks.length).toBe(1)
+
+    const htmlExampleToggleLink = exampleToggleLinks[0]
+    expect(htmlExampleToggleLink.nodeName.toLowerCase()).toBe('a')
+    expect(htmlExampleToggleLink.getAttribute('href')).toBe(`#${exampleId}_html`)
+    expect(htmlExampleToggleLink.getAttribute('aria-controls')).toBe(`${exampleId}_html`)
+    expect(htmlExampleToggleLink.text).toBe('HTML')
+  })
+
+  test('should set the iframe height when it is supplied', () => {
+    const frameHeight = 256
+    const document = documentFactory(Object.assign({ height: frameHeight }, parameters), options)
+    const exampleFrame = document.getElementById(`${exampleId}_frame`)
+    expect(parseInt(exampleFrame.height)).toBe(256)
+  })
+
+  test('Should include the escaped HTML markup from the examples', () => {
+    const exampleHTMLCode = document.getElementById(`${exampleId}_html`)
+    const fixtureHTML = fs.readFileSync(path.join(fixturePath, 'test.html')).toString()
+    expect(exampleHTMLCode.querySelector('pre code').innerHTML).toEqual(htmlEscape(fixtureHTML))
+  })
+})
+
+describe('When a pattern has a nunjucks example', () => {
+  const parameters = { html: `${exampleId}.html`, nunjucks: `${exampleId}.njk` }
+  const document = documentFactory(parameters, options)
+
+  test('should have a button to show Nunjucks example', () => {
+    const codeExampleTabsContainer = document.getElementsByClassName('app-tabs')[0]
+
+    const exampleToggleLinks = codeExampleTabsContainer.querySelectorAll('li.app-tabs__item a')
+    expect(exampleToggleLinks.length).toBe(2)
+
+    const nunjucksExampleToggleLink = exampleToggleLinks[1]
+    expect(nunjucksExampleToggleLink.nodeName.toLowerCase()).toBe('a')
+    expect(nunjucksExampleToggleLink.getAttribute('href')).toBe(`#${exampleId}_nunjucks`)
+    expect(nunjucksExampleToggleLink.getAttribute('aria-controls')).toBe(`${exampleId}_nunjucks`)
+    expect(nunjucksExampleToggleLink.text).toBe('Nunjucks')
+  })
+  // ToDo: Example Nunjucks code container element
+  test('\u001b[33;1mTODO: Should needs a better way to include the Nunjcks macro code for the example\u001b[0m', () => {
+    const exampleCode = document.getElementById(`${exampleId}_nunjucks`)
+    const fixtureCode = fs.readFileSync(path.join(fixturePath, 'test.njk')).toString()
+    expect(exampleCode.querySelector('pre code').innerHTML).toEqual(nunjucksEscape(fixtureCode))
+  })
+})
+
+describe('When a pattern has Welsh content', () => {
+  const savedGlobals = JSON.parse(JSON.stringify(options.globals))
+  const parameters = { html: `${exampleId}.html` }
+  options.globals = Object.assign({ hasWelsh: true }, options.globals)
+  const document = documentFactory(parameters, options)
+
+  afterAll(() => {
+    options.globals = savedGlobals
+  })
```

Comment:
Rather than changing the options.global and the resetting it, is it work just cloning it like the savedGlobals and just using that? Was thinking `const optionsWithWelsh = JSON.parse(JSON.stringify(options.globals))` Just a thought

Reply:
```const options = {
  globals: { filepath: 'test-component/index.njk' }
}

//Set at the top of page
const cloneObj = (o) => JSON.parse(JSON.stringify(options.globals))

//In test
const newOptions = cloneObj(options)
const optionsWithWelsh = Object.assign(newOptions, { hasWelsh: true })```

Reply:
Makes sense, the saved options is just a clone after all